### PR TITLE
Build OpenSearch in CD workflow in order to build security plugin

### PR DESCRIPTION
**Pull request by cliu123**
##  opensearch-security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__

 4. __Why these changes are required?__ 

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -11,14 +11,25 @@ jobs:
 
     steps:
 
-    - name: Set up JDK 11
+    - name: Set up JDK 14
       uses: actions/setup-java@v1
       with:
-        java-version: 11.0.x
+        java-version: 14.0.x
         server-id: sonatype-nexus-staging
```
Comment (vrozov) [inline .github/workflows/cd.yml:18]: Is this still necessary?

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -41,42 +52,43 @@ jobs:
...
-        aws s3 cp --quiet $deb ${s3_prefix}${deb_outfile}
-
-    - name: Upload Artifacts to Maven Central
-      env:
-        SONATYPE_USER: ${{ secrets.SONATYPE_USER }}
-        SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
-      run: |
-        gpg --batch --import --no-tty <(echo -e "${{ secrets.PGP_PRIVATE_KEY }}")
-        mvn -B deploy -Padvanced -Prelease -DskipTests -Dgpg.passphrase=${{ secrets.PGP_PASSPHRASE }}
+# TODO: Update the following configuration commented out once the new CI/CD process of OpenSearch is confirmed.
```
Comment (vrozov) [inline .github/workflows/cd.yml:55]: Delete. It is not clear what will be the approach to publish to maven. In any case we have source control if this needs to be restored.

File: .github/workflows/cd.yml (hunk 1 above)
Comment (dblock) [reply .github/workflows/cd.yml:18]: We have so far failed to start using JDK 11 in OpenSearch core, but you are free to do so here.

Comment (cliu123) [reply .github/workflows/cd.yml]: > Is this still necessary?

Not necessary. Removed.

Comment (cliu123) [reply .github/workflows/cd.yml]: > We have so far failed to start using JDK 11 in OpenSearch core, but you are free to do so here.

This is upgrade from JDK 11 to JDK 14 here.

File: .github/workflows/cd.yml (hunk 2 above)
Comment (cliu123) [reply .github/workflows/cd.yml]: Done

File: .github/workflows/cd.yml (hunk 1 above)
Comment (vrozov) [reply .github/workflows/cd.yml]: I do not refer to JDK change from 11 to 14 that we did for opendistro branches as well (see #1226) and that was supposed to be done as part of #1131 (@cliu123, @debjanibnrj @palashhedau) . My comment applies to `server-id`.

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -11,13 +11,21 @@ jobs:
...
+        java-version: 14.0.x
+
+    - name: Checkout OpenSearch
+      uses: actions/checkout@v2
+      with:
+        repository: 'opensearch-project/OpenSearch'
+        path: OpenSearch
+        ref: '1.0'
+
+    - name: Build OpenSearch
```
Comment (vrozov) [inline .github/workflows/cd.yml:26]: Should this step be after `Cache Maven packages` step?

Comment (cliu123) [reply .github/workflows/cd.yml:26]: Shouldn't OpenSearch artifact built be cached? 
If moving `Build OpenSearch` step and putting it after `Cache Maven packages` step, both of `Build OpenSearch` and `Build` steps will be after `Cache Maven packages` step. Shouldn't the build outputs be cached?

Comment (vrozov) [reply .github/workflows/cd.yml:26]: What is the point of caching OpenSearch maven packages when they are rebuild anyway?

Comment (cliu123) [reply .github/workflows/cd.yml]: Agree. In this case, only downloaded dependencies should be cached. There is no point of caching build outputs. 
Moved. Thanks!

**Review by vrozov — APPROVED** [review]

**Review by palashhedau — APPROVED** [review]

**Pull request merged**